Exerscise_3/main.cpp

```diff
@@ -1,25 +1,26 @@
-#include <istream>
-#include "Cource.h"
-#include "Exam.h"
-#include "Teacher.h"
-#include "Student.h"
 #include <iostream>
 #include <fstream>
-#include <sstream>
-#include <map>
+#include <vector>
 #include <algorithm>
+#include <sstream>
+#include "Course.h"
+#include "Exam.h"
+#include "Teacher.h"
+#include "Student.h"
+
+
 
 template<class T>
-void fill_up_container(std::istringstream &iss, std::map<int, T> &container){
-    T inst(iss);
-    container.insert(std::pair<int, T>(inst.getId(),inst));
+void fill_up_container(std::istringstream &iss, std::vector<std::pair<int, IRecord*>> &container){
+    IRecord* inst = new T(iss);
+    container.emplace_back(inst->getId(),inst);
 }
 
 void make_containers(std::ifstream &file,
-                     std::map <int, Cource> &cource_map,
-                     std::map <int, Exam> &exam_map,
-                     std::map <int, Teacher> &teacher_map,
-                     std::map <int, Student> &student_map){
+                     std::vector<std::pair<int, IRecord*>> &cource_vector,
```

Comment:
Do you really need all of those 4 containers?
If you remove all of those containers, you will ease the reading part (i.e. you won't have to check for the row's mark)